sendgrid.go

```diff
@@ -26,19 +26,38 @@ type SGClient struct {
 }
 
 // NewSendGridClient will return a new SGClient.
-func NewSendGridClient(apiUser, apiPwd string) *SGClient {
+func NewSendGridClient(apiUserOrKey string, apiKey ...string) *SGClient {
```

Comment:
I agree with @deckarep - avoid making the assumption that the user will be passing things correctly. I'd suggest creating an `Auth` struct that can later be passed down to `.New`.